benchbuild/source/base.py

```diff
@@ -61,43 +67,67 @@ def to_str(*variants: Variant) -> str:
     return ",".join([str(i) for i in variants])
 
 
-class ISource(abc.ABC):
+class Fetchable(Protocol):
 
-    @abc.abstractproperty
+    @property
+    def key(self) -> str:
+        """
+        Return the source's key property.
+
+        This provides you with a key component that identifes a single source.
+        It should (no guarantee) be unique among all sources for this project.
+
+        While this make no further assumption, but a good candidate is a
+        file-system name/path.
+        """
+        ...
```

Comment:
Should there be any code here instead of ellipsis?
If nothing should be done, I would prefer
```suggestion
        pass
```

Reply:
As discussed offline: I (try) to use ellipsis on type stubs/protocols, else 'pass'.